src/write_gcm.c

```diff
@@ -574,19 +581,22 @@ static int wg_curl_get_or_post(char *response_buffer,
   curl_result = curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response_code);
   write_ctx.data[0] = 0;
   if (response_code >= 400) {
-    if (!silent_failures) {
+    if (!wg_curl_flag_enabled(flags, WG_CURL_FLAG_SILENT_FAILURES)) {
       WARNING("write_gcm: Unsuccessful HTTP request %ld: %s",
               response_code, response_buffer);
     }
     result = -2;
     goto leave;
   }
 
+  // Buffer overflow.
   if (write_ctx.size < 2) {
-    ERROR("write_gcm: wg_curl_get_or_post: The receive buffer overflowed.");
-    DEBUG("write_gcm: wg_curl_get_or_post: Received data is: %s",
-        response_buffer);
-    goto leave;
+    if (!wg_curl_flag_enabled(flags, WG_CURL_FLAG_IGNORE_BUFFER_OVERFLOW)) {
+      ERROR("write_gcm: wg_curl_get_or_post: The receive buffer overflowed.");
```

Comment:
I would never ERROR here. I think this should be a WARNING log with DEBUG logging actually dumping out the entire content of the buffer each time (i.e. pull the DEBUG bit out of the conditional all together).

Reply:
+1 to both `WARNING` and dumping the buffer via `DEBUG`. I would also suggest logging this message via `DEBUG` in the `else` branch.